### ISSUE TITLE
Add horse riding ($10)

### ISSUE THREAD

**Issue by bennasar99** [cuberite/cuberite#703]
Currently horses act normally when a player is on them, you can't ride them.

[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=5386245)](https://www.bountysource.com/issues/5386245-add-horse-riding?utm_source=5386245&utm_medium=shield&utm_campaign=ISSUE_BADGE)

**Comment by JuliB15** [comment]
What is the problem that causes that nobody made it in two years?

**Comment by NiLSPACE** [comment]
Because it's not a game-breaking feature that we're missing. We have more important features and bugs to work on.

**Comment by JuliB15** [comment]
I'm sorry. Is it possible to help with easy bugs without knowing the whole Cuberite-complex?

**Comment by LogicParrot** [comment]
@JuliB15 Yes, the issues which are marked "easy" do not need a lot of experience to fix. See [this](https://github.com/cuberite/cuberite/issues?q=is%3Aopen+is%3Aissue+label%3Aeasy).

**Comment by PureTryOut** [comment]
Pushing this a bit, is anyone still interested in adding this? It's quite a big missing feature for proper survival servers, and will keep people away from Cuberite (and choose Spigot or the likes instead).

**Comment by SphinxC0re** [comment]
I don't think this is about *interest* rather than time :smile: but you're free to push.

**Comment by PureTryOut** [comment]
I would if I could ;)

But maybe this should be marked as a higher priority? It's pretty important for survival servers.

**Comment by SphinxC0re** [comment]
It's already included in the milestone *Good Survival* so it's on the track :smile:

**Comment by PureTryOut** [comment]
It's been there for 2 years already haha. Not sure if it makes any difference.

**Comment by tonibm19** [comment]
I think pig riding was mostly (or partly) working.
I remember coding somerhing for it.

**Comment by tonibm19** [comment]
https://github.com/cuberite/cuberite/pull/1115
Yep, editing the issue.

**Comment by PureTryOut** [comment]
Awesome, good job! I assume parts of this code can also be used for horse riding?

**Comment by tonibm19** [comment]
probably
I remember trying to code it and getting it (partly and buggy) working.
For example, you had to always look up in order to jump blocks with the horse.
Also, you couldn't jump using space (that's more advanced)

**Comment by PureTryOut** [comment]
Just a note, I've setup a $10 bounty for this issue on Bountysource.

**Comment by PureTryOut** [comment]
Did some horse riding today: 2 and a bit years after creating this issue, and 3.5 years after being added to Minecraft, horses are still really, *really*, **really**, ***really*** buggy.

A list of stuff:

* When trying to tame a horse you get kicked off like it's supposed too. However once you tamed the horse, hearts should come of the horse to show you tamed it succesfully. This doesn't happen
* Horses don't accept any food whatsoever. Not when wild, not when tamed. If given food when wild it's supposed to make it easier tameable
* Once tamed, you can't open it's inventory by right-clicking the horse. If you want to put a saddle on it, you have to hold the saddle and right-click the horse. Although this is nice, it needs an inventory like vanilla. http://minecraft.gamepedia.com/File:13w21a-new-HorseUI-inventory.png for normal horses, https://hydra-media.cursecdn.com/minecraft.gamepedia.com/thumb/7/73/Donkey_UI.png/300px-Donkey_UI.png for donkeys
* Horses can float! 
* Hearts spawn with

**Comment by PureTryOut** [comment]
Did some horse riding today: 2 and a bit years after creating this issue, and 3.5 years after being added to Minecraft, horses are still really, *really*, **really**, ***really*** buggy.

A list of stuff:

* When trying to tame a horse you get kicked off like it's supposed too. However once you tamed the horse, hearts should come of the horse to show you tamed it succesfully. This doesn't happen
* Horses don't accept any food whatsoever. Not when wild, not when tamed. If given food when wild it's supposed to make it easier tameable
* Once tamed, you can't open it's inventory by right-clicking the horse. If you want to put a saddle on it, you have to hold the saddle and right-click the horse. Although this is nice, it needs an inventory like vanilla, examples:
![Normal horses ](https://hydra-media.cursecdn.com/minecraft.gamepedia.com/thumb/c/ce/13w21a-new-HorseUI-inventory.png/300px-13w21a-new-HorseUI-inventory.png?version=105f608fa5fe75cfd6bc6919d55c3adb) Normal horses
![donkeys](https://hydra-media.cursecdn.com/minecraft.gamepedia.com/thumb/7/73/Donkey_UI.png/300px-Donkey_UI.png) Donkeys
* Horses can float! ![Gif](https://gfycat.com/PastFrenchHoneycreeper)
* Hearts *always* spawn with 15 hearts (instead of quite random, it's almost never 15 in vanilla). Also, it seems to spawn with more but being damaged. 
![screenshot_2016-11-21_23-55-07](https://cloud.githubusercontent.com/assets/1565836/20504075/f20c0d84-b045-11e6-8024-8b0c470e5265.png)

* Horses can swim with a player on their back, but they shouldn't be able too. It should start sinking and when fully underwater kick the player off to float back up
* Ears act... strange... ![Gif](https://gfycat.com/MasculineValidAardwolf)
* They can't jump at all
* Every horse is just as quick: really quick

**Comment by mathiascode** [comment]
[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=5386245)](https://www.bountysource.com/issues/5386245-add-horse-riding?utm_source=5386245&utm_medium=shield&utm_campaign=ISSUE_BADGE)

**Comment by PureTryOut** [comment]
I just re-tested it, and it actually seems to work fairly well now. I was able to tame a horse by trying to ride it (and get kicked off a few times) and although the heart particle effect was missing, I'm now able to open the horse inventory.
I could put on a saddle and actually control the horse.

However, although the horse could jump when standing still, it didn't manage to jump when walking/running. It gets it front legs in the air but that's it. When standing still the height of the jump didn't seem to change between horses.
